extensions/datasource/deployment/src/main/java/io/quarkus/datasource/deployment/devservices/DevServicesDatasourceProcessor.java

```diff
@@ -77,20 +78,21 @@ DevServicesDatasourceResultBuildItem launchDatabases(
         //if not and the DB's have already started we just return
         if (databases != null) {
             boolean restartRequired = false;
-            if (!restartRequired) {
-                for (Map.Entry<String, String> entry : cachedProperties.entrySet()) {
-                    if (!Objects.equals(entry.getValue(),
-                            trim(ConfigProvider.getConfig().getOptionalValue(entry.getKey(), String.class).orElse(null)))) {
-                        restartRequired = true;
-                        break;
-                    }
+            Map<String, String> newDatasourceConfigs = ConfigProvider.getConfig()
+                    .unwrap(SmallRyeConfig.class)
+                    .getOptionalValues("quarkus.datasource",
+                            String.class, String.class)
+                    .orElse(Map.of());
+            for (Map.Entry<String, String> entry : cachedProperties.entrySet()) {
+                String newValue = newDatasourceConfigs.get(entry.getKey());
+                if (!Objects.equals(entry.getValue(), trim(newValue))) {
+                    restartRequired = true;
+                    break;
                 }
             }
             if (!restartRequired) {
-                //devservices properties may have been added
-                for (var name : ConfigProvider.getConfig().getPropertyNames()) {
-                    if (name.startsWith("quarkus.datasource.") && name.contains(".devservices.")
-                            && !cachedProperties.containsKey(name) && !EXCLUDED_PROPERTIES.contains(name)) {
```

Comment:
I don't think the excluded properties are necessary anymore, as we get all the datasource properties with `SmallRyeConfig#getOptionalValues`

Reply:
> We may need to adjust the content of `EXCLUDED_PROPERTIES`, though, because I suppose property keys are now relative to `quarkus.datasource.`?

Yes, the path to the `Map` is not included in the keys, so the keys are everything after `quarkus.datasource`.